redisinsight/api/src/modules/database-recommendation/repositories/local.database.recommendation.repository.ts

```diff
@@ -56,6 +60,10 @@ export class LocalDatabaseRecommendationRepository extends DatabaseRecommendatio
     );
     this.eventEmitter.emit(RecommendationEvents.NewRecommendation, [recommendation]);
 
+    const database = await this.databaseService.get(entity?.databaseId)
+
+    this.analytics.sendCreatedRecommendationEvent(recommendation, database)
+
```

Comment:
I found that you assume that entity might be null (`entity?.databaseId`)
if so - I would say that we have 1 of 2 problems:
1. Are you sure `await this.databaseService.get(entity?.databaseId)` will not throw Not Found error
2. if there is no Not Found error (from 1 option) I could see that inside `sendCreatedRecommendationEvent` there is no check for null and if at some point null will be passed inside it will throw an error